TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -474,5 +474,150 @@ private SymbolReference CreateTableOrAliasName(CodeElementsParser.TableOrAliasNa
                 new SymbolType[] { SymbolType.SqlIdentifier, SymbolType.SqlIdentifier });
             return aliasName;
         }
+        private SqlVariable CreateSqlVariable(CodeElementsParser.SqlVariableContext context)
+        {
+            if (context.hostVariable() != null)
+            {
+                return CreateSqlHostVariable(context.hostVariable());
+            }
+            //TODO Add other conditions when adding new  Sql Variable Types 
+            return null;
+        }
+
+        public GetDiagnosticsStatement CreateGetDiagnosticsStatement(CodeElementsParser.GetDiagnosticsStatementContext context)
+        {
+            var isCurrent = context.SQL_CURRENT() != null ? new SyntaxProperty<bool>(true, ParseTreeUtils.GetFirstToken(context.SQL_CURRENT())) : null;
+            var isStacked = context.stacked() != null ? new SyntaxProperty<bool>(true, ParseTreeUtils.GetFirstToken(context.stacked())) : null;
+            Information requestedInformation = null;
+            if (context.statementInformationClauses() != null)
+            {
+                requestedInformation = CreateStatementInformation(context.statementInformationClauses());
+            }
+            else if (context.conditionInformationClause() != null)
+            {
+                requestedInformation = CreateConditionInformationClause(context.conditionInformationClause());
+            }
+            else if (context.combinedInformationClause() != null)
+            {
+                requestedInformation = CreateCombinedInformationClause(context.combinedInformationClause());
+            }
+            return new GetDiagnosticsStatement(isCurrent, isStacked, requestedInformation);
+        }
+
+        private StatementInformation CreateStatementInformation(CodeElementsParser.StatementInformationClausesContext context)
+        {
+            var assignments = context.statementInformationClause().Select(CompositeInformationAssignment).Cast<InformationAssignment>().ToList();
```

Comment:
- The Cast operation should not be necessary, the create method used in the Select could return List<InformationAssignment>
- This will always create `CompositeInformationAssignment` so you'll have to check that `DB2_GET_DIAGNOSTICS_DIAGNOSTICS` or `DB2_SQL_NESTING_LEVEL` are used alone (not mixed with other item names)

Reply:
I propose to verify that in the checker .